components/org.wso2.carbon.identity.policy.password/src/main/java/org/wso2/carbon/identity/policy/password/pattern/registry/impl/DefaultPasswordPatternPolicy.java

```diff
@@ -0,0 +1,177 @@
+                    return pattern.matcher(CharBuffer.wrap(password)).matches();
+                }
+            },
+
+            new BaseRule("Password needs at least a lower case character") {
+
+                private final Pattern pattern = Pattern.compile(".*[a-z].*");
+
+                @Override
+                public boolean passRule(char[] password) {
+                    return pattern.matcher(CharBuffer.wrap(password)).matches();
+
+                }
+            },
+
+            new BaseRule("Password needs at least a number") {
+
+                private final Pattern pattern = Pattern.compile(".*[0-9].*");
+
+                @Override
+                public boolean passRule(char[] password) {
+                    return pattern.matcher(CharBuffer.wrap(password)).matches();
+                }
+            },
+
+            new BaseRule("Password needs at least a special character") {
+
+                private final Pattern pattern = Pattern.compile(".*[-+_!@#$%^&*.,?=].*");
+
+                @Override
+                public boolean passRule(char[] password) {
+                    return pattern.matcher(CharBuffer.wrap(password)).matches();
+                }
+            },
+    };
+
+    List<PasswordRule> effectiveRules = new ArrayList<>();
+
+    public DefaultPasswordPatternPolicy() {
+
+        PasswordPolicyBean passwordPolicyBean = new PasswordPolicyBean();
+
+        if (passwordPolicyBean.getMinLength() > 0) {
+            effectiveRules.add(availableRules[0]);
+        }
+
+        if (passwordPolicyBean.getMaxLength() > 0) {
+            effectiveRules.add(availableRules[1]);
+        }
+
+        if (passwordPolicyBean.isIncludeUpperCase()) {
+            effectiveRules.add(availableRules[2]);
+        }
+
+        if (passwordPolicyBean.isIncludeLowerCase()) {
+            effectiveRules.add(availableRules[3]);
+        }
+
+        if (passwordPolicyBean.isIncludeNumbers()) {
+            effectiveRules.add(availableRules[4]);
+        }
+
+        if (passwordPolicyBean.isIncludeSymbols()) {
+            effectiveRules.add(availableRules[5]);
+        }
+    }
+
+    @Override
+    public ValidationResult enforce(Object... args) {
+
+        char[] password = (char[]) args[0];
+        StringBuilder errorMessage = new StringBuilder();
+        ValidationResult validationResult = new ValidationResult();
+
+        //Give first priority to regex pattern
+        PasswordPolicyBean passwordPolicyBean = new PasswordPolicyBean();
+        if (!passwordPolicyBean.getRegex().isEmpty()) {
+            Pattern pattern = Pattern.compile(passwordPolicyBean.getRegex());
+            if (pattern.matcher(CharBuffer.wrap(password)).matches()) {
+                validationResult.setSuccess(true);
+                return validationResult;
+            }
+            validationResult.setSuccess(false);
+            validationResult.setMessage("Password your entered does not adhere to password policy");
```

Comment:
Password *you* entered. --> Entered password does not adhere to password policy.